### PR TITLE
Update binary name to pyroscope

### PR DESCRIPTION
**Pull request by simonswine**
Also updates makefile targets and some folder names

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
           node-version: 18.15.0
           cache: yarn
       - name: Build image Phlare
-        run: make docker-image/phlare/build "BUILDX_ARGS=--cache-from=type=gha --cache-to=type=gha"
+        run: make docker-image/pyroscope/build "BUILDX_ARGS=--cache-from=type=gha --cache-to=type=gha"
```
Comment (kolesnikovae) [inline .github/workflows/test.yml:87]: `.goreleaser` needs some fixes as well but I'm not sure if it is supposed to be part of this step. Same for helm chart

**Review by cyriltovena — APPROVED** [review]
LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (simonswine) [reply .github/workflows/test.yml:87]: Will try to get weekly builds going next, there is some more work there yes

Comment (simonswine) [reply .github/workflows/test.yml:87]: Most recent commit fixed `make deploy`, we should not touch the helm chart for now

**Pull request merged**